stakedex_sdk/tests/test_main.rs

```diff
@@ -83,18 +65,663 @@ fn test_swap_via_stake_unknown_token() {
+    test_swap_via_stake(msol::ID, cogentsol::ID, None);
+}
+
+#[test]
+fn test_swap_via_stake_msol_daosol() {
+    test_swap_via_stake(msol::ID, daosol::ID, None);
+}
+
+#[test]
+fn test_swap_via_stake_msol_jitosol() {
+    test_swap_via_stake(msol::ID, jitosol::ID, None);
+}
+
+#[test]
+fn test_swap_via_stake_msol_jsol() {
+    test_swap_via_stake(msol::ID, jsol::ID, None);
+}
+
+#[test]
+fn test_swap_via_stake_msol_lainesol() {
+    test_swap_via_stake(msol::ID, lainesol::ID, None);
+}
+
+#[test]
+fn test_swap_via_stake_msol_risksol() {
+    test_swap_via_stake(msol::ID, risksol::ID, None);
+}
+
+#[test]
+fn test_swap_via_stake_msol_scnsol() {
+    test_swap_via_stake(msol::ID, scnsol::ID, None);
+}
+
+#[test]
+fn test_swap_via_stake_msol_esol() {
+    test_swap_via_stake(msol::ID, esol::ID, None);
+}
+
+#[test]
+fn test_swap_via_stake_msol_stsol() {
+    test_swap_via_stake(msol::ID, stsol::ID, None);
+}
+
+// stsol to xsol
+#[test]
+fn test_swap_via_stake_stsol_bsol() {
+    test_swap_via_stake(stsol::ID, bsol::ID, None);
+}
+
+#[test]
+fn test_swap_via_stake_stsol_cogentsol() {
+    test_swap_via_stake(stsol::ID, cogentsol::ID, None);
+}
+
+#[test]
+fn test_swap_via_stake_stsol_daosol() {
+    test_swap_via_stake(stsol::ID, daosol::ID, None);
+}
+
+#[test]
+fn test_swap_via_stake_stsol_jitosol() {
+    test_swap_via_stake(stsol::ID, jitosol::ID, None);
+}
+
+#[test]
+fn test_swap_via_stake_stsol_jsol() {
+    test_swap_via_stake(stsol::ID, jsol::ID, None);
+}
+
+#[test]
+fn test_swap_via_stake_stsol_lainesol() {
+    test_swap_via_stake(stsol::ID, lainesol::ID, None);
+}
+
+#[test]
+fn test_swap_via_stake_stsol_risksol() {
+    test_swap_via_stake(stsol::ID, risksol::ID, None);
+}
+
+#[test]
+fn test_swap_via_stake_stsol_scnsol() {
+    test_swap_via_stake(stsol::ID, scnsol::ID, None);
+}
+
+#[test]
+fn test_swap_via_stake_stsol_esol() {
+    test_swap_via_stake(stsol::ID, esol::ID, None);
+}
+
+#[test]
+fn test_swap_via_stake_stsol_msol() {
+    test_swap_via_stake(stsol::ID, msol::ID, None);
+}
+
+fn test_swap_via_stake(input_mint: Pubkey, output_mint: Pubkey, amount: Option<u64>) {
+    // let user_keypair = read_keypair_file("../test-key.json").unwrap();
+    // let user_pk = user_keypair.pubkey();
     let whale_pk = Pubkey::from_str(WHALE).unwrap();
+
+    let source_token_account = get_associated_token_address(&whale_pk, &input_mint);
+    let source_balance = RPC
+        .get_token_account_balance(&source_token_account)
+        .map_err(|err| {
+            println!("Could not swap {} to {}", input_mint, output_mint);
+            err
+        })
+        .unwrap();
+
+    let destination_token_account = get_associated_token_address(&whale_pk, &output_mint);
+    let destination_balance = RPC
+        .get_token_account_balance(&destination_token_account)
+        .map_err(|err| {
+            println!("Could not swap {} to {}", input_mint, output_mint);
+            err
+        })
+        .unwrap();
```

Comment:
this is causing a lot failures since WHALE doesnt have a lot of the xSOL accounts right? Yeah I think we just exclude those test cases for now. WHALE has
- mSOL
- stSOL
- scnSOL
- jSOL
- wrapped SOL

You can take a look at what he has on https://solana.fm/address/9uyDy9VDBw4K7xoSkhmCAm8NAFCwu4pkF6JeHUCtVKcX?cluster=mainnet-qn1 btw

Reply:
Also i think the panicking caused by the `unwraps()` here is resulting in `Once instance has previously been poisoned` errors for all subsequent test cases